### PR TITLE
Bane AI

### PR DESCRIPTION
**Pull request by tralezab**
### "Bane?"
image

## Why It's Good For The Game

We take Gotham from the corrupt! The rich! The oppressors of generations who have kept you down with myths of opportunity, and we give it back to you...the people...Gotham is yours. None shall interfere, do as you please!

## Changelog
:cl:
add: Bane AI
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
@github-actions
Do you feel in Charge?

**Comment by wesoda25** [comment]
ooohh thats what that dude is from

**Comment by Qustinnus** [comment]
wtf wesoda

**Comment by wesoda25** [comment]
what i knew i recognized him from somewhere but it's been a while since i watched the movie

**Comment by Qustinnus** [comment]
smh how did u not know CIA

**Comment by tralezab** [comment]
image

**Comment by Pugie** [comment]
can you add banepoetry?
[https://www.youtube.com/watch?v=nFzl0_Qeb84](url)
It's _beautiful._

**Comment by Pugie** [comment]
> what i knew i recognized him from somewhere but it's been a while since i watched the movie

I forgot too over a year ago and had to search 'cia' through the discord to jog my memory it was extremely painful.

**Comment by optimumtact** [comment]
> 
> 
> You currently have a negative Fix/Feature pull request delta of -6. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by Qustinnus** [comment]
@ oranges

image

**Comment by tralezab** [comment]
WE BROKE THE BAT
image

File: code/datums/ai/bane/bane_behaviors.dm [hunk 1]
```diff
@@ -4,23 +4,23 @@
...
-	"The powerful will be ripped from their decadent nests, and cast out into the cold world that we know and endure.",
-	"It would be extremely painful.",
-	"As I terrorize Gotham I will feed its people hope to poison their souls.",
-	"The Bomb is armed. The bomb is mobile. The identity of the triggerman is a mystery.",
-	"You think darkness is your ally. But you merely adopted the dark.",
-	"The shadows betray you, because they belong to me...",
-	"I will show you where I've made my home while preparing to bring justice.",
-	"I will break you.",
-	"Your punishment must be more severe.",
+		"They expect one of us in the wreckage, brother.",
```
Comment (optimumtact) [inline code/datums/ai/bane/bane_behaviors.dm:7]: strings framework

**Comment by cacogen** [comment]
CAI when

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Load lines with the string framework

**Comment by tralezab** [comment]
done

File: code/datums/ai/bane/bane_controller.dm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+/datum/ai_controller/bane/SelectBehaviors(delta_time)
+	current_behaviors = list()
+	var/mob/living/batman = blackboard[BB_BANE_BATMAN]
+	if(!batman)
+		for(var/mob/living/possibly_the_dark_knight in oview(7, pawn))
+			if(IS_DEAD_OR_INCAP(possibly_the_dark_knight)) //I HAVE BROKEN THE BAT
+				continue
+			blackboard[BB_BANE_BATMAN] = possibly_the_dark_knight
+			batman = possibly_the_dark_knight
```
Comment (lordpidey) [inline code/datums/ai/bane/bane_controller.dm:28]: Add a break here, so that we early return.

**Review by lordpidey — CHANGES_REQUESTED** [review]

**Comment by CPTANT** [comment]
Batman and Bane are copyrighted material, they should not be mentioned by name in the code.

**Comment by Qustinnus** [comment]
That is not how copyright works.

**Review by Cyberboss — APPROVED** [review]

**Comment by 81Denton** [comment]
> Batman and Bane are copyrighted material, they should not be mentioned by name in the code.

EYA2waOXsAAEUlu

**Pull request merged**